pkg/controller/registry/reconciler/reconciler.go

```diff
@@ -150,5 +150,12 @@ func Pod(source *v1alpha1.CatalogSource, name string, image string, labels map[s
 			},
 		},
 	}
+	if len(source.Spec.Secrets) > 0 {
+		pod.Spec.ImagePullSecrets = []v1.LocalObjectReference{
+			{
+				Name: source.Spec.Secrets[0],
```

Comment:
Per https://github.com/operator-framework/api/blob/master/crds/operators.coreos.com_catalogsources.yaml#L110 the secrets field is an array, I'm not sure we have explicitly documented that the first secret name in the array is the secret for the catalog pod specifically. I think it makes sense, but it could break existing users of the `secrets` field who rely on it to copy secrets from the OLM namespace to the operator's namespace for a variety of uses. So we have to be careful about backwards compatibility with a change like this

Reply:
@exdx exactly what I had in mind too. Which is why I think it's better to accept this secret with a new `spec.imageSecret` field, for which I have opened up a PR here https://github.com/operator-framework/api/pull/79

Reply:
> I'm not sure we have explicitly documented that the first secret name in the array is the secret for the catalog pod specifically.

We should actually be adding all the secrets rather than just the first one.